### PR TITLE
Changed in charge configuration, switched role into list

### PR DESCRIPTION
**Pull request by kkrampa**
@czue Shouldn't be merged before this one https://github.com/dimagi/logistics/pull/221 will be deployed but it's ready to review.

### CODE REVIEW

File: custom/ewsghana/models.py [hunk 1]
```diff
@@ -68,8 +69,6 @@ def update_toggle(self):
             STOCK_AND_RECEIPT_SMS_HANDLER.set(self.domain, True, NAMESPACE_DOMAIN)
 
 
-class AlertsSent(models.Model):
```
Comment (kkrampa) [inline custom/ewsghana/models.py]: Not used.

File: custom/ewsghana/static/ewsghana/ko/ewsghana.users_view_model.js [hunk 2]
```diff
@@ -43,21 +43,31 @@ ko.bindingHandlers.select2 = {
...
+    this.optionsText = this.full_name + " (" + this.locationName + ")";
+}
 
+function UsersViewModel(sms_users, district_in_charges, location_id, submit_url) {
+    var self = this;
     self.locationId = location_id;
     self.submitUrl = submit_url;
 
-    self.users = ko.observableArray(sms_users);
+    self.users = ko.observableArray(sms_users.map(function(user) { return new User(user)}));
```
Comment (dimagimon) [inline custom/ewsghana/static/ewsghana/ko/ewsghana.users_view_model.js]: Missing semicolon.

File: custom/ewsghana/static/ewsghana/ko/ewsghana.users_view_model.js [hunk 3]
```diff
@@ -43,21 +43,31 @@ ko.bindingHandlers.select2 = {
...
 
-    self.visibleUsers = ko.observableArray(ko.utils.arrayFilter(self.users(), function (user) {
-        return !user.in_charge;
-    }));
+    self.districtInCharges = ko.observableArray(district_in_charges.map(function(user) { return new User(user)}));
```
Comment (dimagimon) [inline custom/ewsghana/static/ewsghana/ko/ewsghana.users_view_model.js]: Missing semicolon.

File: custom/ewsghana/static/ewsghana/ko/ewsghana.users_view_model.js [hunk 4]
```diff
@@ -43,21 +43,31 @@ ko.bindingHandlers.select2 = {
...
 
-    self.inCharges = ko.observableArray(ko.utils.arrayFilter(self.users(), function (user) {
-        return user.in_charge;
-    }));
+    self.inCharges = ko.computed(function() {
+        return ko.utils.arrayFilter(self.districtInCharges(), function (user) {
+            return user.in_charge();
+        })
```
Comment (dimagimon) [inline custom/ewsghana/static/ewsghana/ko/ewsghana.users_view_model.js]: Missing semicolon.

**Comment by czue** [comment]
build is failing
```
======================================================================
FAIL: test_parse_sms_user_json (custom.ewsghana.tests.test_apis.ApisTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/dimagi/commcare-hq/custom/ewsghana/tests/test_apis.py", line 60, in test_parse_sms_user_json
    self.assertEqual(smsuser.supply_point.location_id, 591)
AssertionError: 620 != 591
----------------------------------------------------------------------
```

**Comment by kkrampa** [comment]
@czue Fixed

File: custom/ewsghana/models.py [hunk 5]
```diff
@@ -68,8 +69,6 @@ def update_toggle(self):
...
-    create_date = models.DateTimeField(editable=False)
-    alert_type = models.TextField()
-    alert_text = models.TextField()
-    supply_point_id = models.CharField(max_length=10)
+class FacilityInCharge(models.Model):
```
Comment (czue) [inline custom/ewsghana/models.py]: is there a reason we couldn't use the roles in the couch model for this?

Comment (czue) [inline custom/ewsghana/models.py]: nvm, see it's because we want to query it efficiently

File: custom/ewsghana/views.py [hunk 6]
```diff
@@ -304,10 +308,8 @@ def stockouts_product(request, domain):
 def configure_in_charge(request, domain):
```
Comment (czue) [inline custom/ewsghana/views.py]: do we want to wrap this in an `atomic` decorator?

**Comment by czue** [comment]
A couple other thoughts/questions:

1. Is there code that needs to get added to keep the in charge in sync if you change the custom user data? Or are we not supporting that.
2. Do we need to add logic to the domain delete signal catcher to delete the new models?

(also not mergeable anymore - sorry for delayed review)

**Comment by kkrampa** [comment]
@czue 
1.It's good question. I need to ask Mike what we should do. Probably when we remove in charge role from particular user we also need to delete all records for that user from SQL table.
2. Yeah, definitely. I will add this.

File: custom/ewsghana/models.py (hunk 5 above)
Comment (kkrampa) [inline custom/ewsghana/models.py]: No, it's not for efficiency. Now when we assign incharge role to particular user, he becomes in charge in location that he is assigned. This PR changes this behavior. Role in charge doesn't mean that user is in charge for some location but it means that he can be selected as in charge for some location. I hope that now is clear. 

Here is the spec written by Mike:
http://manage.dimagi.com/default.asp?176370#980093

Comment (czue) [inline custom/ewsghana/models.py]: ah thanks that's helpful

**Comment by kkrampa** [comment]
@czue updated

**Comment by czue** [comment]
:+1: when tests pass (and thanks for adding tests).

@emord is there a process we need to do when new migrations are added between now and django 1.7?

**Comment by emord** [comment]
@czue I'd say no. https://github.com/dimagi/commcare-hq/pull/7579#discussion_r35224099

I'll take over keeping the django17 branch up with the migrations on master

**Comment by czue** [comment]
sounds good - thanks

**Pull request merged**